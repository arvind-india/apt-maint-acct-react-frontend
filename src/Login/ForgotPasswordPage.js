import React from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import {
          Form,
          Button,
          FormGroup,
          FormText,
          Input,
          Label,
          Col
        } from 'reactstrap'

import { userActions, alertActions } from '../_actions'


export class ForgotPassword extends React.Component {
  constructor(props) {
    super(props)
    // reset login status
    this.props.logout()
    this.props.clearAlert()
    // this.props.dispatch(userActions.logout())
    // this.props.dispatch(alertActions.clear())  // clear alert messages from other pages
    this.state = {
      email: '',
      submitted: false
    }
    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }
  handleChange(e) {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }
  handleSubmit(e) {
    e.preventDefault()
    this.setState({ submitted: true })
    const { email } = this.state
    // const { dispatch } = this.props
    if(email) {
      // dispatch(userActions.forgotPassword(email))
      this.props.forgotPassword(email)
    }
  }
  render() {
    const { alert } = this.props
    return (
      <div>
        <h2 align="center">Forgot Password</h2>
        {alert.message && <div className={`alert ${alert.type}`}>{alert.message}</div>}
        <Form id="forgotPasswordForm" onSubmit={this.handleSubmit}>
          { this.emailId() }
          { this.buttons() }
        </Form>
      </div>
    )
  }

  emailId() {
    const { email, submitted } = this.state
    return <FormGroup row>
      <Label for="email" sm={3}>Email_Id</Label>
      <Col sm={9}>
        <Input
          id="email"
          type="email"
          name="email"
          placeholder="email id here"
          defaultValue={email}
          onChange={this.handleChange.bind(this)}
        />
        {submitted && !email && <FormText color="danger">Email-id is required</FormText>}
      </Col>
    </FormGroup>
  }

  buttons() {
    return <FormGroup row>
      <Label sm={3}></Label>
      <Col smoffset={3} sm={9}>
        <Button type="submit" color="primary" bssize="large" title="Go to App">Submit</Button>
        <Button color="link" title="Go to home">
          <Link to="/home" className="text-danger" >Cancel</Link>
        </Button>
      </Col>
    </FormGroup>
  }

}

function mapStateToProps(state) {
  const { alert } = state
  return {
    alert
  }
}

function mapDispatchToProps(dispatch) {
  return {
    forgotPassword: (email) => {
      dispatch(userActions.forgotPassword(email))
    },
    logout: () => {
      dispatch(userActions.logout())
    },
    clearAlert: () => {
      dispatch(alertActions.clear())
    }
  }
}

// const connectedForgotPasswordPage = connect(mapStateToProps)(ForgotPasswordPage)
// export { connectedForgotPasswordPage as ForgotPasswordPage }

export default connect(mapStateToProps, mapDispatchToProps)(ForgotPassword)
