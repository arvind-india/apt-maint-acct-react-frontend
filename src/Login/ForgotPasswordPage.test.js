// ForgotPasswordPage.test.js

import React from 'react'
import { shallow } from 'enzyme'

import { ForgotPassword } from './ForgotPasswordPage'

const mockProps = {
  alert: {message: "testingAlert"},
  logout: jest.fn(), // mock function
  clearAlert: jest.fn(), // mock function
  forgotPassword: jest.fn()
}
const mockStates = {
  email: 'unitTesting@jest.js',
}
const mockEmail = {
  name: 'email',
  value: "dummy@email.id"
}
const mockResult = [
  mockEmail.value,
]
const component = shallow(<ForgotPassword {...mockProps} />)

describe('ForgotPasswordPage params test', () => {
  it('should render ForgotPasswordPage without crashing', () => {
    component
  })
  it('should render initial layout', () => {
    expect(component.getElements()).toMatchSnapshot()
  })
  it('should have email state', () => {
    expect(component.state('email')).toBeDefined()
  })
  it('should set email state', () => {
    component.setState({email: mockStates.email})
    expect(component.state('email')).toEqual(mockStates.email)
  })
})

describe('ForgotPasswordPage events test', () => {
  it('should call mock forgotPassword function', () => {
    component.find('#forgotPasswordForm').simulate('submit', {preventDefault() {}})
    expect(mockProps.forgotPassword.mock.calls.length).toEqual(1)
  })
  it('should be called with the email in the state as arguments', () => {
    component.find('#email').simulate('change', {target: mockEmail})
    component.find('#forgotPasswordForm').simulate('submit', {preventDefault() {}})
    expect(mockProps.forgotPassword.mock.calls[1]).toEqual(mockResult)
  })
})
