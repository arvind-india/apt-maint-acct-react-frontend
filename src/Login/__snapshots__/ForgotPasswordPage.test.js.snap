// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordPage params test should render initial layout 1`] = `
Array [
  <div>
    <h2
        align="center"
    >
        Forgot Password
    </h2>
    <div
        className="alert undefined"
    >
        testingAlert
    </div>
    <Form
        id="forgotPasswordForm"
        onSubmit={[Function]}
        tag="form"
    >
        <FormGroup
            row={true}
            tag="div"
        >
            <Label
                for="email"
                sm={3}
                tag="label"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                Email_Id
            </Label>
            <Col
                sm={9}
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Input
                    defaultValue=""
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="email id here"
                    tag="p"
                    type="email"
                />
            </Col>
        </FormGroup>
        <FormGroup
            row={true}
            tag="div"
        >
            <Label
                sm={3}
                tag="label"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            />
            <Col
                sm={9}
                smoffset={3}
                tag="div"
                widths={
                    Array [
                        "xs",
                        "sm",
                        "md",
                        "lg",
                        "xl",
                      ]
                }
            >
                <Button
                    bssize="large"
                    color="primary"
                    tag="button"
                    title="Go to App"
                    type="submit"
                >
                    Submit
                </Button>
                <Button
                    color="link"
                    tag="button"
                    title="Go to home"
                >
                    <Link
                        className="text-danger"
                        replace={false}
                        to="/home"
                    >
                        Cancel
                    </Link>
                </Button>
            </Col>
        </FormGroup>
    </Form>
</div>,
]
`;
